app/backend/db/crud/date_datas.py

```diff
@@ -83,27 +93,42 @@ def delete_users_dates_data(
 
         for date_data in dates_data_to_remove:
             db.delete(date_data)
+    
+    db.commit()
 
 def put_data_in_user(
     db: Session, 
-    body=schemas.PutDateDataBody
+    start_date: date,
+    end_date: date = None,
+    state: schemas.AnswerStateTypes = None,
+    reason: str = None,
+    reported_by_id: int = None,
+    reported_time: datetime = None
 ) -> schemas.RangeDatesResponse:
 
-    request = body.dict()
-    if body.reason is not None:
-        request["reason"] = get_reason_by_name(db, body.reason)
+
+    if reason is not None:
+        reason = get_reason_by_name(db, reason)
```

Comment:
just do the reason must
dont allow default value

Reply:
here and everywhere

Reply:
Here is put, I want to allow a change of a single value without have to give the rest